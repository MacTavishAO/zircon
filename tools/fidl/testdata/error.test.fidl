library fidl.test.json;

protocol Example {
    foo(string s) -> (int64 y) error uint32;
};
