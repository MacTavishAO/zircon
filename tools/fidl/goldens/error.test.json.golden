{
  "version": "0.0.1",
  "name": "fidl.test.json",
  "library_dependencies": [],
  "bits_declarations": [],
  "const_declarations": [],
  "enum_declarations": [],
  "experimental_resource_declarations": [],
  "interface_declarations": [
    {
      "name": "fidl.test.json/Example",
      "location": {
        "filename": "error.test.fidl",
        "line": 3,
        "column": 10,
        "length": 7
      },
      "methods": [
        {
          "ordinal": 841977834905137819,
          "name": "foo",
          "location": {
            "filename": "error.test.fidl",
            "line": 4,
            "column": 5,
            "length": 3
          },
          "has_request": true,
          "maybe_request": [
            {
              "type": {
                "kind": "string",
                "nullable": false
              },
              "name": "s",
              "location": {
                "filename": "error.test.fidl",
                "line": 4,
                "column": 16,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 0
              }
            }
          ],
          "maybe_request_payload": "fidl.test.json/SomeLongAnonymousPrefix0",
          "maybe_request_type_shape_v1": {
            "inline_size": 32,
            "alignment": 8,
            "depth": 1,
            "max_handles": 0,
            "max_out_of_line": 4294967295,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "has_response": true,
          "maybe_response": [
            {
              "type": {
                "kind": "identifier",
                "identifier": "fidl.test.json/Example_foo_Result",
                "nullable": false
              },
              "name": "result",
              "location": {
                "filename": "generated",
                "line": 3,
                "column": 1,
                "length": 6
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 0
              }
            }
          ],
          "maybe_response_payload": "fidl.test.json/SomeLongAnonymousPrefix1",
          "maybe_response_type_shape_v1": {
            "inline_size": 40,
            "alignment": 8,
            "depth": 1,
            "max_handles": 0,
            "max_out_of_line": 8,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "is_composed": false
        }
      ]
    }
  ],
  "service_declarations": [],
  "struct_declarations": [
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix0",
      "location": {
        "filename": "error.test.fidl",
        "line": 4,
        "column": 8,
        "length": 10
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "string",
            "nullable": false
          },
          "name": "s",
          "location": {
            "filename": "error.test.fidl",
            "line": 4,
            "column": 16,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 16,
        "alignment": 8,
        "depth": 1,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/Example_foo_Response",
      "location": {
        "filename": "error.test.fidl",
        "line": 4,
        "column": 22,
        "length": 9
      },
      "anonymous": false,
      "members": [
        {
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "y",
          "location": {
            "filename": "error.test.fidl",
            "line": 4,
            "column": 29,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 8,
        "alignment": 8,
        "depth": 0,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": false,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix1",
      "location": {
        "filename": "error.test.fidl",
        "line": 4,
        "column": 22,
        "length": 9
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "fidl.test.json/Example_foo_Result",
            "nullable": false
          },
          "name": "result",
          "location": {
            "filename": "generated",
            "line": 3,
            "column": 1,
            "length": 6
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 24,
        "alignment": 8,
        "depth": 1,
        "max_handles": 0,
        "max_out_of_line": 8,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    }
  ],
  "table_declarations": [],
  "union_declarations": [
    {
      "name": "fidl.test.json/Example_foo_Result",
      "location": {
        "filename": "error.test.fidl",
        "line": 4,
        "column": 22,
        "length": 9
      },
      "maybe_attributes": [
        {
          "name": "Result",
          "value": ""
        }
      ],
      "members": [
        {
          "ordinal": 1,
          "reserved": false,
          "name": "response",
          "type": {
            "kind": "identifier",
            "identifier": "fidl.test.json/Example_foo_Response",
            "nullable": false
          },
          "location": {
            "filename": "generated",
            "line": 1,
            "column": 1,
            "length": 8
          }
        },
        {
          "ordinal": 2,
          "reserved": false,
          "name": "err",
          "type": {
            "kind": "primitive",
            "subtype": "uint32"
          },
          "location": {
            "filename": "generated",
            "line": 2,
            "column": 1,
            "length": 3
          }
        }
      ],
      "strict": true,
      "type_shape_v1": {
        "inline_size": 24,
        "alignment": 8,
        "depth": 1,
        "max_handles": 0,
        "max_out_of_line": 8,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    }
  ],
  "type_alias_declarations": [],
  "declaration_order": [
    "fidl.test.json/Example_foo_Response",
    "fidl.test.json/Example_foo_Result",
    "fidl.test.json/Example"
  ],
  "declarations": {
    "fidl.test.json/Example": "interface",
    "fidl.test.json/SomeLongAnonymousPrefix0": "struct",
    "fidl.test.json/Example_foo_Response": "struct",
    "fidl.test.json/SomeLongAnonymousPrefix1": "struct",
    "fidl.test.json/Example_foo_Result": "union"
  }
}
